src/game/gui/text_render.h

```diff
@@ -77,16 +78,46 @@ typedef struct {
     uint8_t max_lines;
 } text_settings;
 
+typedef struct letter {
+    int x;
+    int y;
+    int offset;
+    int limit;
```

Comment:
```suggestion
    int palette_offset;
    int palette_limit;
```
Might also be possible to make these a smaller (8-bit) integer type, but that sounds like work for another PR.